docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,166 @@
+# EP: Definition of the Fleet and Device API spec
+<!-- this format is inspired by the K8S KEP format https://raw.githubusercontent.com/kubernetes/enhancements/master/keps/NNNN-kep-template/README.md -->
+## Summary
+
+The Device API, and Fleet API, and the dynamics of managing single fleetless
+devices as well as fleets of devices is a fundamental piece of flightctl.
+
+The Fleet API must serve as a way to manage groups of devices while still allowing
+for the management of individual devices in a way that we can integrate with other
+management systems.
+
+## Motivation
+
+We need agreement on the API design direction and the semantics of the API to be able to
+implement and grow the flightctl service.
+
+### Goals
+
+* Define the format and behavior of the Device and Fleet APIs
+
+* Allow the management of individual fleetless devices on a smaller scale,
+  as an integration mechanism for services like AAP and ACM.
+
+* Allow the management of fleets of devices on a larger scale, via gitops or
+  API.
+
+### Non-Goals
+
+
+
+## Proposal
+
+### Device API
+
+The device API is divided into multiple endpoints, some user facing, and other device/facing,
+this design facilitates cleaner RBAC control of the endpoint access.
+
+#### `/device/{name}` User/Admin facing API
+
+This endpoint can be accessed by the administrator or external systems to
+manage individual devices, also internally from the fleet specifications
+for devices.
+
+When managing a device through a fleet, flightctl is responsible
+of maintaining the device spec in sync with the fleet spec; in this case
+write access is only possible for the modification of labels and other metadata
+through this API.
+
+The proposed changes include:
+
+* Reincorporating device.spec as a visible part through this fronted API.
+* Adding a managementMode field to the device.spec, this field can be set to "fleet" or "manual".
+* The device.spec.os.image contains a reference to the image to be used for the device.
+* The device.spec.containers contains a list of container patterns to be watched on the device (as in the fleet spec)
+* The device.spec.systemd contains a list of systemd service patterns to be watches on the device (as in the fleet spec)
+* The device.spec.config contains a list of config items
+* The config items could be inline definition of config files (inline)
+* Some config items reference exteral configurations (k8s-secret, gitref- ...)
+* External references (image or config items) must support floating tags (latest, main, v1.0),
+  some external items are actually non-versioned, so are inherently floating.
```

Comment:
How can an ops persona understand
1. What is the current content that the service is offering a device?
2. What is the current content running on a device?

Reply:
This would be the equivalent of your debug devspec command , we should maintain that. I would say that may be we should convert it into

flightctl get device/xxxxxxx --rendered

and convert the output for users into all-yaml for readability.